source/utilities.cc

```diff
@@ -2168,9 +2168,19 @@ namespace aspect
         {
           Point<dim> internal_position = position;
 
-          if (dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()) != nullptr
-              || dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model()) != nullptr
-              || dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()) != nullptr)
+          // In the initial topography ascii data plugin, a component variable value of 1000000
+          // is used to indicate that the conversion from cartesian to spherical coordinates
+          // is unnecessary even for models based on spherical symmetry.
+          // However, we still need to use the actual component value for the lookup.
+          unsigned int use_component = component;
+          if (component == 1000000)
+            use_component = 0;
```

Comment:
Oh, I see. No, this is awkward. Let's not do this.

Reply:
Ok! natural_coordinate conversion provide a much neater solution anyway.